caveclient/infoservice.py

```diff
@@ -341,6 +341,53 @@ def viewer_site(self, datastack_name=None, use_stored=True):
             format_for="raw",
         )
 
+    def image_cloudvolume(self, **kwargs):
+        """Generate a cloudvolume instance based on the image source, using authentication if needed and
+        sensible default values for reading CAVE resources. By default, fill_missing is True and bounded
+        is False. All keyword arguments are passed onto the CloudVolume initialization function, and defaults
+        can be overridden.
+
+        Requires cloudvolume to be installed, which is not included by default.
+        """
+        return self._make_cloudvolume(self.image_source(), **kwargs)
+
+    def segmentation_cloudvolume(self, **kwargs):
+        """Generate a cloudvolume instance based on the segmentation source, using authentication if needed and
+        sensible default values for reading CAVE resources. By default, fill_missing is True and bounded
+        is False. All keyword arguments are passed onto the CloudVolume initialization function, and defaults
+        can be overridden.
+
+        Requires cloudvolume to be installed, which is not included by default.
+        """
+        return self._make_cloudvolume(self.segmentation_source(), **kwargs)
+
+    def _make_cloudvolume(self, cloudpath, **kwargs):
+        try:
+            import cloudvolume
+        except:
+            raise ImportError(
+                "Could not import cloudvolume. Make sure it is installed. See https://pypi.org/project/cloud-volume for more info."
+            )
+
+        use_https = kwargs.pop("use_https", True)
+        bounded = kwargs.pop("bounded", False)
+        fill_missing = kwargs.pop("fill_missing", True)
+
+        if re.search("^graphene", cloudpath):
+            # Authentication header is "Authorization {token}"
+            secrets = {"token": self.session.headers.get("Authorization").split(" ")[1]}
+        else:
+            secrets = None
+
+        cv = cloudvolume.CloudVolume(
+            cloudpath,
+            use_https=use_https,
+            fill_missing=fill_missing,
+            bounded=bounded,
```

Comment:
I wonder if we should just pass all other kwargs to cloud volume rather than assuming people want fill missing and bounded

Reply:
Whoops, I had meant to do that! Added that line now. The only reason those two arguments are singled out is that they create surprise errors if the default values are used.